### ISSUE TITLE
After search inconsistencies

### ISSUE THREAD

**Issue by ymjonline** [fabiospampinato/notable#256]
- **OS Version**: Windows 10
- **Notable Version**: 1.1.0 (there's no "About" menu by the way)

### Current behavior

Upon searching the Title column displays the right files (apparently, haven't checked thoroughly though)
However, upon clicking on Tags and Notebooks, the results are changing in an inconsistent way.

It seems it's trying to keep search results and ignore the fact that you're now trying to filter by tags / notebooks / all notes.  However if you click on a tag that one or more of the notes are assigned to, it will try to only display those and discard the others. 
But still, clicking around shows that this behavior is not consistent.

### Expected behavior

I would expect that, once search results are returned, if I then click on "all notes" / a notebook / a tag, the search is "reset" / cancelled.
Or at least with a consitent and expected behavior.

Then the only way to "exit" the search is to actually select the whole keyword field and delete it.
A little X "button" on the search field would be a nice feature.
Like this
 ![image](https://user-images.githubusercontent.com/46528969/50986247-9aa4f600-1506-11e9-9695-c723eb476751.png)
or that
![image](https://user-images.githubusercontent.com/46528969/50986311-cc1dc180-1506-11e9-80c6-695d18fba852.png)

### Screenshot
![noteable_after_search](https://user-images.githubusercontent.com/46528969/50986523-5403cb80-1507-11e9-8e37-0d5409f63f8e.gif)

**Comment by fabiospampinato** [comment]
I mean you're not resetting the search query so it's still applied. I think it would be weird if the query got cleared automatically.

Maybe there could be an `X` button though.

**Comment by ymjonline** [comment]
> I mean you're not resetting the search query so it's still applied. I think it would be weird if the query got cleared automatically.
> 

yes but then, the results list should remain the same whatever I do, until i cleared or changed search.

or at least be consistent by
- showing no result if I click on a Tag that none of the results relate to
- showing only those notes that relate to the tag/notebook that I click on.

**Comment by fabiospampinato** [comment]
@ymjonline if that's not already the case I think it's a bug, but I would find that unlikely. On your screen the notes are matching because we are using fuzzy matching.

**Comment by ymjonline** [comment]
> @ymjonline if that's not already the case I think it's a bug, but I would find that unlikely. On your screen the notes are matching because we are using fuzzy matching.

the search result itself is ok.

it's the change of search result (or lack thereof) upon clicking on tags after a search which is weird

**Comment by fabiospampinato** [comment]
> it's the change of search result (or lack thereof) upon clicking on tags after a search which is weird

Yeah if the result is the same you won't see any difference in the list.

**Comment by ymjonline** [comment]
> > it's the change of search result (or lack thereof) upon clicking on tags after a search which is weird
> 
> Yeah if the result is the same you won't see any difference in the list.

I do see a difference, and that's precisely the problem.

Either the 
i) the result should remain the same no matter what i do, or
ii) result should reset once I start clicking around on Tags/Notebooks or
iii) result should filter based on which tag/folder I click on: e.g. if I click on Tag "X" and none of the notes from the results have that Tag, then show nothing. If 3 notes out of 10 from the results have that Tag, show those 3

right now it's just not clear what to expect and it changes unconsistently.

**Comment by fabiospampinato** [comment]
@ymjonline what result is returned but shouldn't and vice versa?
